CountlyConnectionManager.m

```diff
@@ -197,17 +197,39 @@ - (void)proceedOnQueue
     [CountlyCommon.sharedInstance startBackgroundTask];
 
     queryString = [self appendRemainingRequest:queryString];
-    queryString = [self appendChecksum:queryString];
+    NSMutableData* pictureUploadData = [self pictureUploadDataForQueryString:queryString];
+
+    if (!pictureUploadData)
+    {
+        queryString = [self appendChecksum:queryString];
+    }
 
     NSString* serverInputEndpoint = [self.host stringByAppendingString:endPoint];
     NSString* fullRequestURL = [serverInputEndpoint stringByAppendingFormat:@"?%@", queryString];
     NSMutableURLRequest* request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:fullRequestURL]];
 
-    NSData* pictureUploadData = [self pictureUploadDataForQueryString:queryString];
     if (pictureUploadData)
     {
+        request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:serverInputEndpoint]];
```

Comment:
We are already creating a URLRequest on line 209 without checking the picture upload data without any checks, i think its better to use if else something like this for line 202-213: 
`NSString* serverInputEndpoint = [self.host stringByAppendingString:endPoint];
if (!pictureUploadData)
    {
        queryString = [self appendChecksum:queryString];
        NSString* fullRequestURL = [serverInputEndpoint stringByAppendingFormat:@"?%@", queryString];
    NSMutableURLRequest* request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:fullRequestURL]];
    }
    else {
        request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:serverInputEndpoint]];
        ....
        ...
        ...
        }
    `